### PR TITLE
feat: Refactor inputsrc and remove INPUTSRC_TYPE from ovms

### PR DESCRIPTION
**Pull request by vli11**
Closes: #291

## PR Checklist

- [x] Added label to the Pull Request for easier discoverability and search
- [x] Commit Message meets guidelines as indicated in the URL https://github.com/intel-retail/automated-self-checkout/blob/main/CONTRIBUTING.md
- [x] Every commit is a single defect fix and does not mix feature addition or changes
- [ ] Unit Tests have been added for new changes
- [ ] Updated Documentation as relevant to the changes
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with repository license and clearly outlined the added dependency.
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide details below)

## What are you changing?
For OVMS workload only, this PR removed manipulated $inputsrc, and added user input $INPUTSRC to env variable for framework and remove $INPUTSRC_TYPE since it is not used.

## Issue this PR will close

close: #291 

## Test Instructions if applicable
- make run-smoke-tests
- the result should be all passed.

### CODE REVIEW

File: run-ovms.sh [hunk 1]
```diff
@@ -54,35 +52,17 @@ elif grep -q "file" <<< "$INPUTSRC"; then
 	arrfilesrc=(${INPUTSRC//:/ })
 	# use vids since container maps a volume to this location based on sample-media folder
 	# TODO: need to pass demux/codec info
-	inputsrc="filesrc location=vids/"${arrfilesrc[1]}" ! qtdemux ! h264parse "
```
Comment (jim-wang-intel) [inline run-ovms.sh:57]: so where is this logic moved to ?   i think we probably add this into gst profile

File: run-ovms.sh [hunk 2]
```diff
@@ -54,35 +52,17 @@ elif grep -q "file" <<< "$INPUTSRC"; then
...
-	inputsrc=$INPUTSRC
-	INPUTSRC_TYPE="USB"
 	TARGET_USB_DEVICE="--device=$INPUTSRC"
 	decode_type="videoconvert ! video/x-raw,format=BGR"
 	pre_process=""
 
 else
-	# rs-serial realsenssrc
-	# TODO need to pass depthalign info
-	inputsrc="realsensesrc cam-serial-number="$INPUTSRC" stream-type=0 align=0 imu_on=false"
```
Comment (jim-wang-intel) [inline run-ovms.sh:74]: same here

File: run-ovms.sh [hunk 3]
```diff
@@ -54,35 +52,17 @@ elif grep -q "file" <<< "$INPUTSRC"; then
...
-    # add realsense color related properties if any
-	if [ "$COLOR_WIDTH" != 0 ]; then
-		inputsrc=$inputsrc" color-width="$COLOR_WIDTH
-	fi
-	if [ "$COLOR_HEIGHT" != 0 ]; then
-		inputsrc=$inputsrc" color-height="$COLOR_HEIGHT
-	fi
-	if [ "$COLOR_FRAMERATE" != 0 ]; then
-		inputsrc=$inputsrc" color-framerate="$COLOR_FRAMERATE
-	fi
```
Comment (jim-wang-intel) [inline run-ovms.sh:84]: same here

**Review by jim-wang-intel — COMMENTED** [review]
some question regarding the changes applied to

File: configs/opencv-ovms/demos/classification/python/entrypoint.sh [hunk 4]
```diff
@@ -5,5 +5,5 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 python3 classification/python/classification_demo.py -m localhost:"$GRPC_PORT"/models/"$CLASSIFICATION_MODEL_NAME" \
-	--label classification/python/labels/"$CLASSIFICATION_LABEL_FILE" -i $inputsrc \
+	--label classification/python/labels/"$CLASSIFICATION_LABEL_FILE" -i $INPUTSRC \
```
Comment (antoniomtz) [inline configs/opencv-ovms/demos/classification/python/entrypoint.sh:8]: I have an open PR that may be affected by this change. Let's just make sure we sync up. 
I'm setting this variable as env var on the run_classification.sh script

Comment (antoniomtz) [reply configs/opencv-ovms/demos/classification/python/entrypoint.sh:8]: Same for all changes below

File: run-ovms.sh [hunk 5]
```diff
@@ -40,53 +40,16 @@ fi
...
-
+re='^[0-9]+$'
+if [[ "$INPUTSRC" != "rtsp*" && "$INPUTSRC" != "file*" && "$INPUTSRC" != *video* && "$INPUTSRC" =~ $re ]]; then
+	echo "assume realsense device..."
+	cameras=`ls /dev/vid* | while read line; do echo "--device=$line"; done`
+	TARGET_GPU_DEVICE=$TARGET_GPU_DEVICE" "$cameras
 elif grep -q "video" <<< "$INPUTSRC"; then
+	echo "assume video device..."
 	# v4l2src /dev/video*
 	# TODO need to pass stream info
```
Comment (jim-wang-intel) [inline run-ovms.sh:51]: let's simplify this logic:
1) first check the video on the first if
2) if not, then check inputsrc to regular expression you already got; if matching the pattern then will assume it is real sense camera
3) else - then ignore...

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]

**Review by antoniomtz — APPROVED** [review]
LGTM. Tested locally.

File: run-ovms.sh (hunk 1 above)
Comment (vli11) [reply run-ovms.sh:57]: moved to run_gst.sh

File: run-ovms.sh (hunk 2 above)
Comment (vli11) [reply run-ovms.sh:74]: moved to run_gst.sh

File: run-ovms.sh (hunk 3 above)
Comment (vli11) [reply run-ovms.sh:84]: moved to run_gst.sh

**Review by brian-intel — APPROVED** [review]

File: configs/opencv-ovms/scripts/run_gst.sh [hunk 6]
```diff
@@ -21,15 +24,53 @@ then
 	rmDocker=
 fi
 
+echo "$PLATFORM"
+if [ "$PLATFORM" == "dgpu" ]; then
```
[the remark this answers is not in the document]
Comment (jim-wang-intel) [reply configs/opencv-ovms/scripts/run_gst.sh:28]: never mind- looks like the other code get-options.sh set this up; so we are good here

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Pull request merged**